test/Ported/Server/Documents/Notifications/ChangesTest.ts

```
@@ -0,0 +1,308 @@
+            await store.maintenance.send(operation);
+
+            const indexChange = await changesList.poll(2000);
+            assert.ok(indexChange);
+
+            assert.strictEqual(indexChange.name, new UsersByName().getIndexName());
+        } finally {
+            observable.off("data", handler);
+        }
+    });
+
+    it("can obtain all index changes", async () => {
+        const changesList = new AsyncQueue<IndexChange>();
+
+        await store.executeIndex(new UsersByName());
+
+        const changes = store.changes();
+        await changes.ensureConnectedNow();
+
+        const observable = changes.forAllIndexes();
+        await observable.ensureSubscribedNow();
+
+        const handler = (change: IndexChange) => changesList.push(change);
+
+        try {
+            observable.on("data", handler);
+            observable.on("error", e => throwError("InvalidOperationException", e.message));
+
+            const operation = new SetIndexesPriorityOperation(new UsersByName().getIndexName(), "Low");
+            await store.maintenance.send(operation);
+
+            const indexChange = await changesList.poll(2000);
+            assert.ok(indexChange);
+
+            assert.strictEqual(indexChange.name, new UsersByName().getIndexName());
+        } finally {
+            observable.off("data", handler);
+        }
+    });
+
+    it("can obtain notification about documents starting with", async () => {
+        const changesList = new AsyncQueue<DocumentChange>();
+
+        const changes = store.changes();
+        await changes.ensureConnectedNow();
+
+        const observable = changes.forDocumentsStartingWith("users/");
+        await observable.ensureSubscribedNow();
+
+        const handler = (change: DocumentChange) => changesList.push(change);
+
+        try {
+            observable.on("data", handler);
+            observable.on("error", e => throwError("InvalidOperationException", e.message));
```

I think throwing inside of the handler won't bubble up to try block, but will silently fail there. Since it's a test you'd need to do `done(err)` or similar.

---

I've added message.